packages/mc-scripts/src/commands/deployment-previews-push.ts

```diff
@@ -0,0 +1,266 @@
+          ),
+      ]
+        .filter(Boolean)
+        .join('\n'),
+      validate: (value) => value && value.length > 2,
+    });
+    deploymentUrl = url as string;
+  }
+  if (!validateUrl(deploymentUrl)) {
+    console.log(chalk.red('Invalid deployment URL.'));
+    console.log(chalk.red('Aborted.'));
+    return;
+  }
+
+  if (options.dryRun) {
+    const message = existingDeploymentPreview
+      ? `The existing deployment preview with alias "${deploymentAlias}" would have been updated with this url: "${deploymentUrl}".`
+      : `A new deployment preview with alias "${deploymentAlias}" would have been created with this url: "${deploymentUrl}".`;
+    console.log(`\n${message}\n`);
+    return;
+  }
+
+  const mcDeploymentPreviewsLink = getMcUrlLink({
+    mcApiUrl,
+    organizationId: fetchedCustomApplication.organizationId,
+    customEntityId: fetchedCustomApplication.application.id,
+    isCustomView,
+  });
+  if (existingDeploymentPreview) {
+    await updateCustomApplicationDeploymentPreview({
+      mcApiUrl,
+      organizationId: fetchedCustomApplication.organizationId,
+      applicationIdentifier,
+      deploymentId: existingDeploymentPreview.id,
+      data: { url: deploymentUrl },
+    });
+
+    console.log(
+      chalk.green(
+        `Deployment preview with alias "${deploymentAlias}" updated successfully with URL "${deploymentUrl}".`
+      )
+    );
+    console.log(
+      chalk.green(
+        `You can access the deployment previews at: ${mcDeploymentPreviewsLink}`
+      )
+    );
+  } else {
+    await createCustomApplicationDeploymentPreview({
+      mcApiUrl,
+      organizationId: fetchedCustomApplication.organizationId,
+      applicationIdentifier,
+      applicationId,
+      data: { alias: deploymentAlias, url: deploymentUrl },
+    });
+
+    console.log(
+      chalk.green(
+        `Deployment preview with alias "${deploymentAlias}" created successfully with URL "${deploymentUrl}".`
+      )
+    );
+    console.log(
+      chalk.green(
+        `You can access the deployment previews at: ${mcDeploymentPreviewsLink}`
+      )
+    );
+  }
+}
+
+async function run(options: TCliCommandPushDeploymentPreviewOptions) {
+  const applicationConfig = processConfig();
+  const localCustomEntityData: CustomApplicationData | CustomViewData =
+    applicationConfig.data;
+  const { mcApiUrl, applicationIdentifier } = applicationConfig.env;
+  const isCustomView = isCustomViewData(localCustomEntityData);
+
+  console.log(`Using Merchant Center environment "${chalk.green(mcApiUrl)}".`);
+  console.log();
+
+  const isSessionValid = credentialsStorage.isSessionValid(mcApiUrl);
+  if (!isSessionValid) {
+    throw new Error(
+      `You don't have a valid session. Please, run the "mc-scripts login" command to authenticate yourself.`
+    );
+  }
+
+  if (isCustomView) {
+    throw new Error(
+      'Deployments previews are not supported for Custom Views yet.'
+    );
+  }
```

Comment:
Do you envision deployment previews for Custom Views?

Maybe we can rephrase this just to avoid false expectations.

```suggestion
  if (isCustomView) {
    throw new Error(
      'Deployments previews are not supported for Custom Views.'
    );
  }
```

Reply:
Yes I do.

That's something we considered as part of the Custom Views features but left out of the first iteration to keep the scope contained.

Reply:
Message updated in this commit: aa8ac4684f43bce4c5332169e38bd039964e6212